libsycl/cmake/Modules/FetchUnifiedRuntime.cmake

```
@@ -0,0 +1,120 @@
+#===============================================================================
+# Fetches Unified Runtime used by SYCL language runtime to abstract SYCL open 
+# standard and vendor heterogeneous offload interfaces.
+#
+# This will in time be replaced by the new LLVM Offload interface.
+#
+# Unified Runtime is Apache 2.0 license with LLVM exceptions. 
+#
+#===============================================================================
+
+option(LIBSYCL_UR_BUILD_TESTS "Build tests for UR" OFF)
+
+set(UR_BUILD_TESTS "${LIBSYCL_UR_BUILD_TESTS}" CACHE BOOL "" FORCE)
+# UR tests require the examples to be built
+set(UR_BUILD_EXAMPLES "${LIBSYCL_UR_BUILD_TESTS}" CACHE BOOL "" FORCE)
+
+if("level_zero" IN_LIST LIBSYCL_ENABLE_BACKENDS)
+  set(UR_BUILD_ADAPTER_L0 ON)
+endif()
+if("cuda" IN_LIST LIBSYCL_ENABLE_BACKENDS)
+  set(UR_BUILD_ADAPTER_CUDA ON)
+endif()
+if("hip" IN_LIST LIBSYCL_ENABLE_BACKENDS)
+  set(UR_BUILD_ADAPTER_HIP ON)
+endif()
+if("opencl" IN_LIST LIBSYCL_ENABLE_BACKENDS)
+  set(UR_BUILD_ADAPTER_OPENCL ON)
+endif()
+
+# Disable errors from warnings while building the UR.
+# And remember origin flags before doing that.
+set(CMAKE_CXX_FLAGS_BAK "${CMAKE_CXX_FLAGS}")
+if(WIN32)
+  append("/WX-" CMAKE_CXX_FLAGS)
+  append("/WX-" CMAKE_C_FLAGS)
+  # Unified runtime build fails with /DUNICODE
+  append("/UUNICODE" CMAKE_CXX_FLAGS)
+  append("/UUNICODE" CMAKE_C_FLAGS)
+else()
+  append("-Wno-error" CMAKE_CXX_FLAGS)
+  append("-Wno-error" CMAKE_C_FLAGS)
+endif()
+
+if(NOT FETCHCONTENT_SOURCE_DIR_UNIFIED-RUNTIME)
+  find_package(unified-runtime)
+  if(unified-runtime_FOUND)
+    message ("STATUS: unified-runtime is found")
```

nit

```suggestion
    message (STATUS "Found system install of unified-runtime")
```

or something like that

---

https://github.com/sergey-semenov/llvm-project/pull/4/commits/142c2243f7a1a27d1191a97580b4b4f2637a3beb